<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Jimmy Christensen</author>
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>484</width>
    <height>466</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
    <horstretch>1</horstretch>
    <verstretch>1</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>OldSkoolGravityGame - Launcher</string>
  </property>
  <property name="windowIcon">
   <iconset resource="std.qrc">
    <normaloff>:/icon/icon.png</normaloff>:/icon/icon.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QGridLayout" name="gridLayout_3">
    <property name="margin">
     <number>0</number>
    </property>
    <item row="0" column="0">
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="sizeConstraint">
       <enum>QLayout::SetMinAndMaxSize</enum>
      </property>
      <item>
       <widget class="QTabWidget" name="tabWidget">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="gameTab">
         <attribute name="title">
          <string>Game</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QPushButton" name="btnStart">
            <property name="toolTip">
             <string>Start the game!</string>
            </property>
            <property name="text">
             <string>Start Osgg!</string>
            </property>
            <property name="icon">
             <iconset resource="std.qrc">
              <normaloff>:/icon/icon.png</normaloff>:/icon/icon.png</iconset>
            </property>
            <property name="iconSize">
             <size>
              <width>32</width>
              <height>32</height>
             </size>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBoxLevels">
            <property name="title">
             <string>Levels</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_4">
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <property name="spacing">
              <number>0</number>
             </property>
             <item row="0" column="0">
              <layout class="QHBoxLayout" name="horizontalLayout">
               <item>
                <widget class="QTabWidget" name="tabLvl">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="minimumSize">
                  <size>
                   <width>0</width>
                   <height>0</height>
                  </size>
                 </property>
                 <property name="maximumSize">
                  <size>
                   <width>201</width>
                   <height>16777215</height>
                  </size>
                 </property>
                 <property name="currentIndex">
                  <number>0</number>
                 </property>
                 <widget class="QWidget" name="tab">
                  <attribute name="title">
                   <string>Standard</string>
                  </attribute>
                  <layout class="QGridLayout" name="gridLayout">
                   <item row="0" column="0">
                    <layout class="QVBoxLayout" name="verticalLayout_7">
                     <item>
                      <layout class="QHBoxLayout" name="horizontalLayout_3">
                       <item>
                        <widget class="QLabel" name="label_2">
                         <property name="text">
                          <string>Starting Level:</string>
                         </property>
                        </widget>
                       </item>
                       <item>
                        <widget class="QSpinBox" name="numStartLevel"/>
                       </item>
                      </layout>
                     </item>
                     <item>
                      <widget class="QListWidget" name="lstStdLvl">
                       <property name="sortingEnabled">
                        <bool>false</bool>
                       </property>
                      </widget>
                     </item>
                    </layout>
                   </item>
                  </layout>
                 </widget>
                 <widget class="QWidget" name="tab_2">
                  <attribute name="title">
                   <string>Custom</string>
                  </attribute>
                  <layout class="QGridLayout" name="gridLayout_8">
                   <item row="0" column="0">
                    <widget class="QPushButton" name="btnCreate">
                     <property name="text">
                      <string>Create</string>
                     </property>
                    </widget>
                   </item>
                   <item row="0" column="1">
                    <widget class="QPushButton" name="btnEdit">
                     <property name="text">
                      <string>Edit</string>
                     </property>
                    </widget>
                   </item>
                   <item row="1" column="0" colspan="2">
                    <widget class="QListWidget" name="lstCustomLvl"/>
                   </item>
                   <item row="2" column="0">
                    <widget class="QPushButton" name="btnAddCustom">
                     <property name="text">
                      <string>Add</string>
                     </property>
                    </widget>
                   </item>
                   <item row="2" column="1">
                    <widget class="QPushButton" name="btnRemoveCustom">
                     <property name="text">
                      <string>Remove</string>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </widget>
                </widget>
               </item>
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_6">
                 <item>
                  <layout class="QVBoxLayout" name="verticalLayout_5">
                   <item>
                    <widget class="QLabel" name="txtTimesPlayed">
                     <property name="text">
                      <string>timesplayed</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QLabel" name="txtBestTime">
                     <property name="text">
                      <string>besttime</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QLabel" name="txtTimesCompleted">
                     <property name="text">
                      <string>timescompleted</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QLabel" name="txtTimesDead">
                     <property name="text">
                      <string>timeddead</string>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                 <item>
                  <widget class="lvlpreviewview" name="lvlPreview">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                     <horstretch>1</horstretch>
                     <verstretch>1</verstretch>
                    </sizepolicy>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="demoTab">
         <attribute name="title">
          <string>Demoes</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QPushButton" name="btnPlayDemo">
            <property name="text">
             <string>Play Demo!</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <property name="sizeConstraint">
             <enum>QLayout::SetDefaultConstraint</enum>
            </property>
            <item>
             <widget class="QLabel" name="label_5">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>LevelFile:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="txtDemoLevel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btnDemoBrowse">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="autoFillBackground">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>...</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QListWidget" name="lstDemo"/>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_3">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <attribute name="title">
          <string>Settings</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_8">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <layout class="QGridLayout" name="gridLayout_2">
              <property name="sizeConstraint">
               <enum>QLayout::SetMinimumSize</enum>
              </property>
              <item row="0" column="1">
               <widget class="QLineEdit" name="txtExecPath">
                <property name="text">
                 <string>./osgg</string>
                </property>
               </widget>
              </item>
              <item row="0" column="2">
               <widget class="QToolButton" name="btnBrowseExecPath">
                <property name="text">
                 <string comment="Browse for the executable">...</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="label_3">
                <property name="text">
                 <string>osgg data path:</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QLineEdit" name="txtDataDir">
                <property name="text">
                 <string>./</string>
                </property>
               </widget>
              </item>
              <item row="1" column="2">
               <widget class="QToolButton" name="btnBrowseDataDir">
                <property name="text">
                 <string comment="Browse for the data directory">...</string>
                </property>
               </widget>
              </item>
              <item row="0" column="0">
               <widget class="QLabel" name="label_4">
                <property name="text">
                 <string>osgg executable:</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="title">
             <string>Screen</string>
            </property>
            <layout class="QFormLayout" name="formLayout">
             <property name="horizontalSpacing">
              <number>4</number>
             </property>
             <property name="verticalSpacing">
              <number>4</number>
             </property>
             <item row="0" column="0">
              <widget class="QCheckBox" name="chFullScreen">
               <property name="toolTip">
                <string>Run game in fullscreen mode</string>
               </property>
               <property name="text">
                <string>Fullscreen</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QCheckBox" name="chShowFps">
               <property name="toolTip">
                <string>Show how many fps the game runs at, this value should be close to 60 for optimal play.</string>
               </property>
               <property name="text">
                <string>showfps</string>
               </property>
               <property name="checked">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QCheckBox" name="chVsync">
               <property name="toolTip">
                <string>Turn on Vsync for smooth gameplay.</string>
               </property>
               <property name="text">
                <string>vsync</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QLabel" name="label">
               <property name="text">
                <string>Background color:</string>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QToolButton" name="btnSelectBgColor">
               <property name="toolTip">
                <string/>
               </property>
               <property name="styleSheet">
                <string notr="true">background-color: rgb(0, 0, 0);</string>
               </property>
               <property name="text">
                <string>...</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_2">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="title">
             <string>Sound</string>
            </property>
            <layout class="QFormLayout" name="formLayout_3">
             <item row="0" column="0">
              <widget class="QCheckBox" name="chSound">
               <property name="toolTip">
                <string>Enable in-game sound effects</string>
               </property>
               <property name="text">
                <string>Sound On</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_3">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="title">
             <string>Gameplay</string>
            </property>
            <layout class="QFormLayout" name="formLayout_2">
             <property name="fieldGrowthPolicy">
              <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
             </property>
             <item row="0" column="0">
              <widget class="QCheckBox" name="chIs">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="baseSize">
                <size>
                 <width>0</width>
                 <height>0</height>
                </size>
               </property>
               <property name="toolTip">
                <string>Save progress automatically when you land on a base</string>
               </property>
               <property name="text">
                <string>Save at every platform</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>lvlpreviewview</class>
   <extends>QGraphicsView</extends>
   <header location="global">lvlpreviewview.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="std.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>lstStdLvl</sender>
   <signal>currentRowChanged(int)</signal>
   <receiver>numStartLevel</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>367</y>
    </hint>
    <hint type="destinationlabel">
     <x>149</x>
     <y>257</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
